File: 0121-best-time-to-buy-and-sell-stock/0121-best-time-to-buy-and-sell-stock.cpp
class Solution {
public:
    int maxProfit(vector<int>& prices) {
        int maxProfit=0;
        int minEle=INT_MAX;
        for(int i=0;i<prices.size();i++){
            minEle=min(minEle,prices[i]);
            if(prices[i]-minEle > maxProfit)
                maxProfit=prices[i]-minEle;
        }
        return maxProfit;
    }
};

File: 0121-best-time-to-buy-and-sell-stock/0121-best-time-to-buy-and-sell-stock.java
class Solution {
    public int maxProfit(int[] prices) {
        int min = prices[0],profit = 0;
        for(int i=0;i<prices.length;i++){
            if(prices[i] < min){
                min=prices[i];
            }
            else if(prices[i]-min > profit){
                profit = prices[i]-min;
            }
        }
        return profit;
    }
}